pkg/controller/controllerutil/controllerutil.go

```diff
@@ -91,6 +91,29 @@ func SetControllerReference(owner, controlled metav1.Object, scheme *runtime.Sch
 	return nil
 }
 
+// RemoveControllerReference is a helper method to make sure the given object removes an controller reference to the object provided.
+// This allows you to remove the owner to establish a new owner of the object in a subsequent call.
+func RemoveControllerReference(owner, controlled metav1.Object) error {
+	owners := controlled.GetOwnerReferences()
+	length := len(owners)
+	result := []metav1.OwnerReference{}
+	if length < 1 {
+		return fmt.Errorf("%T does not have any owner references", controlled)
+	}
+	for _, ownerref := range owners {
+		if ownerref.Name == owner.GetName() {
+			continue
+		}
+		result = append(result, ownerref)
+	}
```

Comment:
Could we use `indexOwnerRef` here, find the index (if any) and cut the slice instead of creating a brand new one?

Reply:
I was trying to do that but was failing miserably as my slice foo isn't there.

Also, I didn't want to stomp on situations where we would try to re-slice an index of 1 and be out of bounds by `[index+1]`.  It seemed premature to handle a specific case where we would re-slice for length of 1 instead of allocating a new slice and re-set.  

I can use this method to see if I  can do that then what I'm currently doing.

Reply:
Couple of questions: @vincepri 
1. Re-slicing, I still have to `SetControllerReference` once it's done, do we need to keep the same slice? 
2. This causes me to make more method(s) to get the GVK of the object which doesn't have any exposed method to do so.  The method you are suggesting gets the owner ref from the ownerReferences to do this where that information is available. The `ref.Name == owner.GetName()` isn't available in that function for me to migrate this over to that.

Reply:
Similarly how the other functions handle it sounds reasonable, https://github.com/kubernetes-sigs/controller-runtime/pull/2455/files#diff-049ddcd31ccf7b24e1e222739d4e5129f1ea5583be912bacc6b0fb5251695b12R120-R133

Reply:
Thanks for the suggestion @vincepri  I updated the logic to use that function and updated the unit tests as well.